src/main/java/uk/nhs/hee/tis/trainee/sync/service/PlacementSyncService.java

```diff
@@ -52,14 +70,29 @@ public void syncRecord(Record record) {
     } else {
       repository.save((Placement) record);
     }
+
+    switch (record.getOperation()) {
+      case LOAD:
+      case INSERT:
+      case UPDATE:
+        enrichOrRequestPost(record);
+        break;
+      case DELETE:
+    }
   }
 
-  public Set<Placement> findByPostId(String postId) {
-    return repository.findByPostId(postId);
+  public void enrichOrRequestPost(Record record) {
+    String postId = record.getData().get("postId");
+    Optional<Post> fetchedPost = postSyncService.findById(postId);
+
+    if (fetchedPost.isPresent()) {
+      placementEnricherFacade.enrich((Post)record);
+    } else {
+      postSyncService.request(postId);
+    }
```

Comment:
We're doing too much in this service, the enricher will handle everything for us. Just need to implement the `request` method.